src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Deployments.Core.Extensions;
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    internal class LinterAnalyzer : IBicepAnalyzer
+    {
+        public const string SettingsRoot = "analyzers";
+        public const string AnalyzerName = "bicepcorelinter";
+        public static string LinterEnabledSetting => $"{SettingsRoot}:{AnalyzerName}:enabled";
+        public static string LinterVerboseSetting => $"{SettingsRoot}:{AnalyzerName}:verbose";
+
+        private ConfigHelper configHelper = new ConfigHelper();
+        private ImmutableArray<IBicepAnalyzerRule> RuleSet;
+
+        public LinterAnalyzer()
+        {
+            RuleSet = CreateLinterRules().ToImmutableArray();
+        }
+
+        private bool LinterEnabled => this.configHelper.GetValue(LinterEnabledSetting, true);
+        private bool LinterVerbose => this.configHelper.GetValue(LinterVerboseSetting, true);
+
+        private IEnumerable<IBicepAnalyzerRule> CreateLinterRules()
+        {
+            var ruleTypes = Assembly.GetExecutingAssembly()
+                .GetTypes()
+                .Where(t => typeof(IBicepAnalyzerRule).IsAssignableFrom(t)
+                            && !t.IsInterface && !t.IsAbstract); // exlude the interface and the base class
```

Comment:
>IsInterface [](http://example.com/codeflow?start=34&length=11)

Should we check `IsClass` instead? Can a rule be a struct?

Reply:
done